### ISSUE TITLE
[VoiceOver/Chrome-Safari] Store name link is vocalized but doesn't work

### ISSUE THREAD

**Issue by developpeurweb** [SAP/spartacus#13359]
**Describe the bug**

- While navigating page content using Rotor> Headings, and once a specific store name has been focused, the user is unable to follow the link because it seems code order is inversed.
- Instead of `<a><h2></h2></a>` as it is in most places, there we have `<h2><a></a></h2>`, so both the heading and the link gets vocalized but VoiceOver is unable to follow the link with this combination.

**Tell us the version of Spartacus**
- Library version: 4.0.0

**To Reproduce**
1. Go to Find a Store > View All Stores > Any country
2. Open VoiceOver
3. Open Rotor in VoiceOver and select "Headings" (ctrl+opt+U).
4. Select a heading for a store name so it gets focused.
5. Try to follow the link by pressing Enter key.

**Expected behavior**
The user should be able to focus heading using the rotor and follow the link.

**Screenshots**
![Screen Shot 2021-07-28 at 11 38 40 AM](https://user-images.githubusercontent.com/15113219/128052595-353ba649-5ebe-48ec-b9d8-5e1ebfcf055a.png)

**Desktop (please complete the following information):**

OS: BigSur 11.4

- Browser Chrome Version 92.0.4515.107
- Browser Safari Version 14.1.1 (16611.2.7.1.4)

**Additional context**
- Not an issue for JAWS/Chrome (Win)
- Not an issue for NVDA/Chrome (Win)

**Comment by KateChuen** [comment]
Tested on develop branch of spartacus using installation script
More extensive testing (on other screenreaders and devices) will be done with RC.

When focus is on store name, I can hear the name. 
Also able to go to the store details pa
![FindStore7](https://user-images.githubusercontent.com/45409515/157734917-67827e11-c1a7-4fbc-a146-0bc67706a5b1.png)
![FindStore6](https://user-images.githubusercontent.com/45409515/157734921-ad699ce7-569d-4030-8d70-884ba5295e23.png)
![FindStore5](https://user-images.githubusercontent.com/45409515/157734926-f645210e-4b09-4039-89d4-64d8be7b2b84.png)
ge and it's vocalised.
